### PR TITLE
CoordinateSource pixel_size handling

### PR DESCRIPTION
**Pull request by garrettwrong**
Adds `pixel_size` to initializer for CoordinateSources.  Also add basic test for the explicit, None, and mismatch case.

### CODE REVIEW

File: tests/test_coordinate_source.py [hunk 1]
```diff
@@ -700,6 +734,44 @@ def testCommand(self):
         self.assertTrue(result_star.exit_code == 0)
         self.assertTrue(result_preprocess.exit_code == 0)
 
+    def testPixelSizeWarning(self):
+        """
+        Test source haveing a pixel size that conflicts with the CTFFilter instances.
```
Comment (j-c-c) [inline tests/test_coordinate_source.py]: "having"

File: tests/test_coordinate_source.py [hunk 2]
```diff
@@ -700,6 +734,44 @@ def testCommand(self):
...
+        manual_pixel_size = 0.789
+        src = BoxesCoordinateSource(self.files_box, pixel_size=manual_pixel_size)
+        # Capture and compare warning message
+        with pytest.warns(UserWarning, match=r".*Pixel size mismatch.*"):
+            src.import_relion_ctf(self.relion_ctf_file)
+            np.testing.assert_approx_equal(src.pixel_size, manual_pixel_size)
+
+    def testMultiplePixelSizeWarning(self):
+        """
+        Test source haveing a multiple pixel sizes in CTFFilter instances.
```
Comment (j-c-c) [inline tests/test_coordinate_source.py]: "having" ~a~

File: src/aspire/source/coordinates.py [hunk 3]
```diff
@@ -385,6 +403,33 @@ def _extract_ctf(self, data_block):
         # convert defocus_ang from degrees to radians
         filter_params[:, 3] *= np.pi / 180.0
 
+        # Check pixel_size
+        # Get pixel_sizes from CTFFilters
+        ctf_pixel_sizes = list(set(filter_params[:, 6]))
```
Comment (j-c-c) [inline src/aspire/source/coordinates.py]: Do we need to worry about using `set` on floating points? I'm guessing it wouldn't be a problem that comes up, but since I'm less familiar with how these values are packaged I just wanted to check.

**Review by j-c-c — COMMENTED** [review]
This looks good. Just one questions and a couple typos if you feel like changing them.

File: src/aspire/source/coordinates.py (hunk 3 above)
Comment (garrettwrong) [reply src/aspire/source/coordinates.py]: It does come up. The values are either equivalent as a`float` or not. If they are not, this is what we want to test by using set.

I can probably use `np.unqiue` here instead, but its the same situation. I'll try that since it's already an array...

File: tests/test_coordinate_source.py (hunk 1 above)
Comment (garrettwrong) [reply tests/test_coordinate_source.py]: thanks, i suppose I set myself up for these rushing this together 😆 . I should have used AI 😂 . Fixed both.

Comment (j-c-c) [reply tests/test_coordinate_source.py]: 🤣

**Review by j-c-c — DISMISSED** [review]
LGTM!

File: src/aspire/source/coordinates.py [hunk 4]
```diff
@@ -46,7 +47,17 @@ class CoordinateSource(ImageSource, ABC):
...
+    def __init__(
+        self, files, particle_size, max_rows, B, symmetry_group, pixel_size=None
+    ):
+        """
+        :param files: A list of tuples of the form (path_to_mrc, path_to_coord)
+        :param particle_size: Desired size of cropped particles (will override the size specified in coordinate file)
+        :param max_rows: Maximum number of particles to read. (If `None`, will attempt to load all particles)
+        :param B: CTF envelope decay factor
+        :param symmetry_group: A `SymmetryGroup` object or string corresponding to the symmetry of the molecule.
+        :param pixel_size: Pixel size of the images in angstroms, default `None`.
```
Comment (janden) [inline src/aspire/source/coordinates.py]: We should indicate what `None` means in this context (auto-assign from CTF if possible).

**Review by janden — CHANGES_REQUESTED** [review]
Looks good. Just one thing.

File: src/aspire/source/coordinates.py (hunk 4 above)
Comment (garrettwrong) [reply src/aspire/source/coordinates.py]: Sure I changed the three occurrences to the following

```
        :param pixel_size: Pixel size of the images in angstroms.               
            Default `None` will attempt to infer `pixel_size` from              
            `CTFFilter` objects when available.  
            
```
